File: Assets/My Assets/Scripts/Gameplay/Player/PlayerLocalMP.js
//Controls our ships
var myType : ShipType; //0/Fighter | 1/Harvester
var selected = false;
var allowInput : boolean = true;
var bullet : GameObject; //gameobject
var myRigidbody : Rigidbody2D;
var fireSound : AudioClip;
var explosionSound : AudioClip;
var speed : float = 0.1; //How quickly do we accelerate?
var turnSpeed : float = 1.5; //How fast can we rotate and change orientation?
var cooldown : float = 0.5; //How long until we can fire our weapon?
public var maxHealth : float = 0.1; //How much health can we have if fully healed?
var maxSpeed : float = 250; //How fast can we move?
var fireForce : float = 100; //How fast does our bullet move?
var explosionParticle : GameObject; //Explosion effect when we die :(
private var fireTimer : float = 0; //Cooldown timer. You shouldn't have to screw with this unless something goes terribly wrong!
public var shipHealth : float = maxHealth; //current Health. This should stay private, to help stop weird health modification bugs
var target : GameObject;
function Start(){
	myRigidbody = this.GetComponent(Rigidbody2D);
	shipHealth = maxHealth;
}
function maintainSpeed(){ //Slow down when above speed parameters
	if(selected == true){
		if(myRigidbody.velocity.magnitude >= maxSpeed){
			myRigidbody.drag = 20;
		}
		else{
			myRigidbody.drag = 0.1;
		}
	}
	else{
		if(myRigidbody.velocity.magnitude >= maxSpeed*0.5){ //Bottleneck speed when we aren't controlling it!
			myRigidbody.drag = 20;
		}
		else{
			myRigidbody.drag = 0.1;
		}
	}
}
function Fire(){ //Shoot a bullet, or catch an Asteroid in a Harvester. No Caching! :(
	if(fireTimer <= 0.01){
		fireTimer = cooldown;
		if(myType == 0){ //Fighter
			var myBullet = Instantiate(bullet,this.transform.position, Quaternion.identity);
			Physics2D.IgnoreCollision(myBullet.GetComponent(BoxCollider2D),this.GetComponent(BoxCollider2D));
			myBullet.GetComponent(Rigidbody2D).velocity = this.transform.up*fireForce;
			myBullet.GetComponent(Rigidbody2D).velocity += this.GetComponent(Rigidbody2D).velocity; //0.5 is more correct, but makes you feel like you shot at an angle at high speeds.
			this.GetComponent(AudioSource).PlayOneShot(fireSound, 0.5);
		}
	}
}
function shipInput(){ //Check for button presses, act accordingly.
	//Movement
	if(allowInput == true){
		this.transform.localRotation.eulerAngles.z -= Input.GetAxisRaw("HorizontalP2") * turnSpeed *Time.deltaTime*14;
		myRigidbody.AddForce(this.transform.up*Input.GetAxisRaw("VerticalP2")*speed); //2D physics
		if(Input.GetButton("SlowP2")){
			myRigidbody.velocity*=0.9;
		}
		if(Input.GetButton("FireP2")){
			Fire();
		}
	}
}
function weaponTimer(){ //Iterate through the shooting cooldown.
	if(fireTimer > 0){
		fireTimer-=Time.deltaTime;
	}
	if(fireTimer < 0){
		fireTimer=0;
	}
}
function Die(){ //Spawn an explosion, and delete the GameObject.
	var explosion = Instantiate(explosionParticle,this.transform.position,Quaternion.identity);
	var mySound = new GameObject(); //Spawn explosion audio GameObject. Could've sworn there was a better way for this.
	mySound.transform.position = this.transform.position;
	mySound.AddComponent(AudioSource);
	mySound.GetComponent(AudioSource).spatialBlend = 1;
	mySound.GetComponent(AudioSource).PlayOneShot(explosionSound, 1);
	Destroy(mySound, 10); //Cleanup!
	Destroy(explosion, 20);
	Destroy(this.gameObject); //This should be changed, probably.
}
function OnCollisionEnter2D(mycol : Collision2D){ //Check if we're being hurt/shot at
	if(mycol.collider.tag == "SpikeEnemy"){
		shipHealth-=1;
	}
	if(mycol.collider.tag == "Bullet"){
		shipHealth-=0.5;
		if(selected == true){
			GameObject.FindGameObjectWithTag("MainCamera").BroadcastMessage("ScreenShake");
		}
		Destroy(mycol.gameObject);
	}
}
function clearSelection(){
	selected = false;
}
function Update () { //Called every frame
	if(shipHealth <= 0){
		Die();
	}
	if(selected == true){ //Are we controlling this ship?
		myRigidbody.isKinematic = false;
		shipInput();
		this.transform.tag = "PlayerLocalMP";
	}
	weaponTimer();
}
public function getMaxHealth(){
	return maxHealth;
}
public function getShipHealth(){
	return shipHealth;
}

File: Assets/My Assets/Scripts/Gameplay/Player/PlayerLocalMP.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class PlayerLocalMP : MonoBehaviour
{
    //Controls our ships
    public ShipType myType; //0/Fighter | 1/Harvester
    public bool selected;
    public bool allowInput;
    public GameObject bullet; //gameobject
    public Rigidbody2D myRigidbody;
    public AudioClip fireSound;
    public AudioClip explosionSound;
    public AudioClip damagedSound;
    public AudioClip slowSound;
    public float speed; //How quickly do we accelerate?
    public float turnSpeed; //How fast can we rotate and change orientation?
    public float cooldown; //How long until we can fire our weapon?
    public float maxHealth; //How much health can we have if fully healed?
    public float maxSpeed; //How fast can we move?
    public float fireForce; //How fast does our bullet move?
    public GameObject explosionParticle; //Explosion effect when we die :(
    private float fireTimer; //Cooldown timer. You shouldn't have to screw with this unless something goes terribly wrong!
    public float shipHealth; //current Health. This should stay private, to help stop weird health modification bugs
    public GameObject target;
    public virtual void Start()
    {
        this.myRigidbody = (Rigidbody2D) this.GetComponent(typeof(Rigidbody2D));
        this.shipHealth = this.maxHealth;
    }

    public virtual void maintainSpeed() //Slow down when above speed parameters
    {
        if (this.selected == true)
        {
            if (this.myRigidbody.velocity.magnitude >= this.maxSpeed)
            {
                this.myRigidbody.drag = 20;
            }
            else
            {
                this.myRigidbody.drag = 0.1f;
            }
        }
        else
        {
            if (this.myRigidbody.velocity.magnitude >= (this.maxSpeed * 0.5f)) //Bottleneck speed when we aren't controlling it!
            {
                this.myRigidbody.drag = 20;
            }
            else
            {
                this.myRigidbody.drag = 0.1f;
            }
        }
    }

    public virtual void Fire() //Shoot a bullet, or catch an Asteroid in a Harvester. No Caching! :(
    {
        if (this.fireTimer <= 0.01f)
        {
            this.fireTimer = this.cooldown;
            if (this.myType == (ShipType) 0) //Fighter
            {
                GameObject myBullet = UnityEngine.Object.Instantiate(this.bullet, this.transform.position, Quaternion.identity);
                Physics2D.IgnoreCollision((BoxCollider2D) myBullet.GetComponent(typeof(BoxCollider2D)), (BoxCollider2D) this.GetComponent(typeof(BoxCollider2D)));
                ((Rigidbody2D) myBullet.GetComponent(typeof(Rigidbody2D))).velocity = this.transform.up * this.fireForce;
                ((Rigidbody2D) myBullet.GetComponent(typeof(Rigidbody2D))).velocity = ((Rigidbody2D) myBullet.GetComponent(typeof(Rigidbody2D))).velocity + ((Rigidbody2D) this.GetComponent(typeof(Rigidbody2D))).velocity; //0.5 is more correct, but makes you feel like you shot at an angle at high speeds.
                ((AudioSource) this.GetComponent(typeof(AudioSource))).PlayOneShot(this.fireSound, 0.5f);
            }
        }
    }

    public virtual void shipInput() //Check for button presses, act accordingly.
    {
        //Movement
        if (this.allowInput == true)
        {
            this.transform.localRotation.eulerAngles.z = this.transform.localRotation.eulerAngles.z - (((Input.GetAxisRaw("HorizontalP2") * this.turnSpeed) * Time.deltaTime) * 14);
            this.myRigidbody.AddForce((this.transform.up * Input.GetAxisRaw("VerticalP2")) * this.speed); //2D physics
            if (Input.GetButton("SlowP2"))
            {
                this.myRigidbody.velocity = this.myRigidbody.velocity * 0.9f;
                if (this.myRigidbody.velocity.magnitude >= 10)
                {
                    ((AudioSource) this.GetComponent(typeof(AudioSource))).clip = this.slowSound;
                    if (((AudioSource) this.GetComponent(typeof(AudioSource))).isPlaying == false)
                    {
                        ((AudioSource) this.GetComponent(typeof(AudioSource))).Play();
                        ((AudioSource) this.GetComponent(typeof(AudioSource))).loop = true;
                    }
                }
            }
            if (Input.GetButton("FireP2"))
            {
                this.Fire();
            }
        }
    }

    public virtual void weaponTimer() //Iterate through the shooting cooldown.
    {
        if (this.fireTimer > 0)
        {
            this.fireTimer = this.fireTimer - Time.deltaTime;
        }
        if (this.fireTimer < 0)
        {
            this.fireTimer = 0;
        }
    }

    public virtual void Die() //Spawn an explosion, and delete the GameObject.
    {
        GameObject explosion = UnityEngine.Object.Instantiate(this.explosionParticle, this.transform.position, Quaternion.identity);
        GameObject mySound = new GameObject(); //Spawn explosion audio GameObject. Could've sworn there was a better way for this.
        mySound.transform.position = this.transform.position;
        mySound.AddComponent(typeof(AudioSource));
        ((AudioSource) mySound.GetComponent(typeof(AudioSource))).spatialBlend = 1;
        ((AudioSource) mySound.GetComponent(typeof(AudioSource))).PlayOneShot(this.explosionSound, 1);
        UnityEngine.Object.Destroy(mySound, 10); //Cleanup!
        UnityEngine.Object.Destroy(explosion, 20);
        UnityEngine.Object.Destroy(this.gameObject); //This should be changed, probably.
    }

    public virtual void OnCollisionEnter2D(Collision2D mycol) //Check if we're being hurt/shot at
    {
        if (mycol.collider.tag == "SpikeEnemy")
        {
            this.shipHealth = this.shipHealth - 1;
        }
        if (mycol.collider.tag == "Bullet")
        {
            this.shipHealth = this.shipHealth - 0.5f;
            if (this.selected == true)
            {
                GameObject.Find("Main Camera2").BroadcastMessage("ScreenShake");
                GameObject mySound = new GameObject(); //Spawn explosion audio GameObject. Could've sworn there was a better way for this.
                mySound.transform.position = this.transform.position;
                mySound.AddComponent(typeof(AudioSource));
                ((AudioSource) mySound.GetComponent(typeof(AudioSource))).spatialBlend = 0;
                ((AudioSource) mySound.GetComponent(typeof(AudioSource))).pitch = ((AudioSource) mySound.GetComponent(typeof(AudioSource))).pitch + (Random.Range(0, 1) * 0.1f);
                ((AudioSource) mySound.GetComponent(typeof(AudioSource))).PlayOneShot(this.damagedSound, 1);
                UnityEngine.Object.Destroy(mySound, 10); //Cleanup!
            }
            UnityEngine.Object.Destroy(mycol.gameObject);
        }
    }

    public virtual void clearSelection()
    {
        this.selected = false;
    }

    public virtual void Update() //Called every frame
    {
        if (this.shipHealth <= 0)
        {
            this.Die();
        }
        if (this.selected == true) //Are we controlling this ship?
        {
            this.myRigidbody.isKinematic = false;
            this.shipInput();
            this.transform.tag = "PlayerLocalMP";
        }
        this.weaponTimer();
    }

    public virtual float getMaxHealth()
    {
        return this.maxHealth;
    }

    public virtual float getShipHealth()
    {
        return this.shipHealth;
    }

    public PlayerLocalMP()
    {
        this.allowInput = true;
        this.speed = 0.1f;
        this.turnSpeed = 1.5f;
        this.cooldown = 0.5f;
        this.maxHealth = 0.1f;
        this.maxSpeed = 250;
        this.fireForce = 100;
        this.shipHealth = this.maxHealth;
    }

}